pino.d.ts

```diff
@@ -13,193 +13,204 @@
 //                 Austin Beer <https://github.com/austin-beer>
 //                 Michel Nemnom <https://github.com/Pegase745>
 //                 Igor Savin <https://github.com/kibertoad>
-// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.0
+//                 James Bromwell <https://github.com/thw0rted>
+// TypeScript Version: 4.4
 
-/// <reference types="node"/>
+import type { EventEmitter } from "events";
+import type { PrettyOptions } from "pino-pretty";
+import type { SonicBoom, SonicBoomOpts } from "sonic-boom";
+import type { WorkerOptions } from "worker_threads";
 
-import { EventEmitter } from "events";
-import { SonicBoom } from "sonic-boom";
 import * as pinoStdSerializers from "pino-std-serializers";
-import { WriteStream } from "fs";
-import { WorkerOptions } from "worker_threads";
 
-export default P;
-export { P as pino }
-export type { P }
 
-type LogDescriptor = Record<string, any>;
-type MessageFormatFunc = (log: LogDescriptor, messageKey: string, levelLabel: string) => string;
+//// Non-exported types and interfaces
 
-/**
- * @param [optionsOrStream]: an options object or a writable stream where the logs will be written. It can also receive some log-line metadata, if the
- * relative protocol is enabled. Default: process.stdout
- * @returns a new logger instance.
- */
-declare function P(optionsOrStream?: P.LoggerOptions | P.DestinationStream): P.Logger;
+type SerializerFn = (value: any) => any;
+type WriteFn = (o: object) => void;
```

Comment:
are you sure you want object and not `Record<string, any>` here?

Reply:
I didn't write that, just moved it around from the old typings.  There are a lot of places where I don't think the types are optimal, but I wasn't going to try to fix them all.